### ISSUE TITLE
Duplicate image sending in specific configuration [SOLVED - PEBCAK]

### ISSUE THREAD

**Issue by uriel1998** [EionRobb/purple-pulsesms#8]
Currently I have a chain of `purple-pulsesms -> bitlbee -> weechat_1 -> relayed (via weechat protocol)  to  weechat_2` on a different box.   This is an edge/unusual case (vaguely necessary in my case), but it produces a pretty strange behavior.

I also use both the web interface and my phone to send images sometimes.

Condition 1: I send an image from my phone, image is recieived once by recipients, regardless of how much of the purple-pulsesms chain above is running.
Condition 2: I send an image from the web interface without any of the purple-pulsesms chain above active. Image is recieved once by recipients.
Condition 3: I send an image from the web interface with the purple-pulsesms STOPPING at "weechat_1".  Image is recieved once by recipients.
Condition 4: I send an image from the web interface with the ENTIRE purple-pulsesms chain active. Image is recieved *twice* by recipients.

Since purple-pulsesms doesn't handle images at all, I'm not sure why this specific configuration causes an issue.  The weechat relay should be (and is) transparent for text messages, with recipients only getting one copy of a text SMS regardless of how I sent it or how much of that chain is active.

Also, a public thank you to my girlfriend for putting up with all the bad dad joke memes I sent her while I was checking those various conditions.  :)

**Comment by EionRobb** [comment]
First off:  there is no such thing as a bad dad joke


That is a really unusual bug!  You're right that I never added in image sending, so I don't quite know what could be causing that.  Can you get a purple debug log out of bitlbee?  Hopefully that sheds some light on things  https://wiki.bitlbee.org/Debugging

**Comment by uriel1998** [comment]
Sure, I'll see what I can get out of it tomorrow.  (Been beating my head against ghostscript all day, and so my "software testing" brain is a bit exhausted.)  Just didn't want you to think I was ignoring you; you've been SO responsive and it's really really cool.  :)

**Comment by uriel1998** [comment]
Whenever I ran the debug console, the behavior did not recur.  So yay?

My best hypothesis at the moment is PEBCAK; I had Monit watching to make sure the bitlbee process was up by watching the PIDfile, and I noticed while testing that it said the service wasn't running (even though I'd manually started it).  I switched Monit to watching the port for bitlbee instead after testing, and reset my inital configuration, and the behavior hasn't recurred.  So I suspect that Monit thought bitlbee wasn't running, and spawned a second copy of bitlbee (or something) producing the error.  

WHY monit didn't say "oh, the PID file is there, it must be running", I have no idea, but if monitoring the port instead makes the behavior go away, I'm happy.  Thank you again for your responsiveness and efforts with this.
